File: TDS/client/damagesys/death.ts
mp.events.add( "playerSpawn", ( player: PlayerMp ) => {
	if ( player == localPlayer ) {
		mp.game.cam.doScreenFadeIn( 2000 );
	}
} );

mp.events.add( "playerDeath", ( player: PlayerMp, reason, killer ) => {
	if ( player == localPlayer ) {
		mp.game.cam.doScreenFadeOut( 2000 );
		mp.game.gameplay.ignoreNextRestart( true );
        mp.game.gameplay.setFadeOutAfterDeath( false );
        scaleformMessage.showWastedMessage();
        mp.game.audio.requestScriptAudioBank( "HUD_MINI_GAME_SOUNDSET", true );  //
        mp.game.audio.playSoundFrontend( -1, "CHECKPOINT_NORMAL", "HUD_MINI_GAME_SOUNDSET", true );  //
	}
} );

File: TDS_Client/Manager/Damage/Death.cs
using RAGE.Game;
using TDS_Client.Default;
using TDS_Client.Manager.Draw.Scaleform;
using TDS_Client.Manager.Event;
using TDS_Client.Manager.Utility;
using Player = RAGE.Elements.Player;

namespace TDS_Client.Manager.Damage
{
    internal static class Death
    {
        public static void PlayerSpawn()
        {
            Cam.DoScreenFadeIn(Settings.ScreenFadeInTimeAfterSpawn);
            Graphics.StopScreenEffect(DEffectName.DEATHFAILMPIN);
            Cam.SetCamEffect(0);
        }

        public static void PlayerDeath(Player player)
        {
            if (player != Player.LocalPlayer)
                return;
            Cam.DoScreenFadeOut(Settings.ScreenFadeOutTimeAfterSpawn);
            Misc.IgnoreNextRestart(true);
            Misc.SetFadeOutAfterDeath(false);
            Audio.PlaySoundFrontend(-1, DAudioName.BED, DAudioRef.WASTEDSOUNDS, true);
            Cam.SetCamEffect(1);
            Graphics.StartScreenEffect(DEffectName.DEATHFAILMPIN, 0, true);

            ScaleformMessage.ShowWastedMessage();

            CustomEventManager.SetDead();
        }
    }
}